dataRetriever/factory/resolverscontainer/shardResolversContainerFactory.go

```diff
@@ -0,0 +1,357 @@
+
+func (srcf *shardResolversContainerFactory) generateHdrResolver() ([]string, []dataRetriever.Resolver, error) {
+	shardC := srcf.shardCoordinator
+
+	//only one shard header topic, for example: shardBlocks_0_META
+	identifierHdr := factory.ShardBlocksTopic + shardC.CommunicationIdentifier(sharding.MetachainShardId)
+
+	peerListCreator, err := topicResolverSender.NewDiffPeerListCreator(srcf.messenger, identifierHdr, emptyExcludePeersOnTopic)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	hdrStorer := srcf.store.GetStorer(dataRetriever.BlockHeaderUnit)
+	resolverSender, err := topicResolverSender.NewTopicResolverSender(
+		srcf.messenger,
+		identifierHdr,
+		peerListCreator,
+		srcf.marshalizer,
+		srcf.intRandomizer,
+		shardC.SelfId(),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(shardC.SelfId())
+	hdrNonceStore := srcf.store.GetStorer(hdrNonceHashDataUnit)
+	resolver, err := resolvers.NewHeaderResolver(
+		resolverSender,
+		srcf.dataPools.Headers(),
+		hdrStorer,
+		hdrNonceStore,
+		srcf.marshalizer,
+		srcf.uint64ByteSliceConverter,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	//add on the request topic
+	_, err = srcf.createTopicAndAssignHandler(
+		identifierHdr+resolverSender.TopicRequestSuffix(),
+		resolver,
+		false)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return []string{identifierHdr}, []dataRetriever.Resolver{resolver}, nil
+}
+
+//------- PeerChBlocks resolvers
+
+func (srcf *shardResolversContainerFactory) generatePeerChBlockBodyResolver() ([]string, []dataRetriever.Resolver, error) {
+	shardC := srcf.shardCoordinator
+
+	//only one intrashard peer change blocks topic
+	identifierPeerCh := factory.PeerChBodyTopic + shardC.CommunicationIdentifier(shardC.SelfId())
+	peerBlockBodyStorer := srcf.store.GetStorer(dataRetriever.PeerChangesUnit)
+
+	peerListCreator, err := topicResolverSender.NewDiffPeerListCreator(srcf.messenger, identifierPeerCh, emptyExcludePeersOnTopic)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resolverSender, err := topicResolverSender.NewTopicResolverSender(
+		srcf.messenger,
+		identifierPeerCh,
+		peerListCreator,
+		srcf.marshalizer,
+		srcf.intRandomizer,
+		shardC.SelfId(),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resolver, err := resolvers.NewGenericBlockBodyResolver(
+		resolverSender,
+		srcf.dataPools.MiniBlocks(),
+		peerBlockBodyStorer,
+		srcf.marshalizer,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	//add on the request topic
+	_, err = srcf.createTopicAndAssignHandler(
+		identifierPeerCh+resolverSender.TopicRequestSuffix(),
+		resolver,
+		false)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return []string{identifierPeerCh}, []dataRetriever.Resolver{resolver}, nil
+}
+
+//------- MetaBlockHeaderResolvers
+
+func (srcf *shardResolversContainerFactory) generateMetablockHeaderResolver() ([]string, []dataRetriever.Resolver, error) {
```

Comment:
and also be consistent in functions name because you have a function generateHdrResolver and function  generateMetablockHeaderResolvers ...use Hdr or Header in all functions names

Reply:
done